qiskit_experiments/library/characterization/spectroscopy.py

```diff
@@ -0,0 +1,124 @@
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+from qiskit.qobj.utils import MeasLevel
+
+from qiskit_experiments.framework import BaseAnalysis, BaseExperiment, Options
+from qiskit_experiments.curve_analysis import ResonanceAnalysis
+
+
+class Spectroscopy(BaseExperiment, ABC):
+    """An abstract class for spectroscopy experiments."""
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default option values used for the spectroscopy pulse.
+
+        Experiment Options:
+            amp (float): The amplitude of the spectroscopy pulse. Defaults to 0.1 and must
+                be between 0 and 1.
+            duration (int): The duration of the spectroscopy pulse. Defaults to 1024 samples.
+            sigma (float): The standard deviation of the flanks of the spectroscopy pulse.
+                Defaults to 256.
+            width (int): The width of the flat-top part of the GaussianSquare pulse.
+                Defaults to 0.
+        """
+        options = super()._default_experiment_options()
+
+        options.amp = 0.1
+        options.duration = 1024
+        options.sigma = 256
+        options.width = 0
+
+        return options
+
+    @classmethod
+    def _default_run_options(cls) -> Options:
+        """Default options values for the experiment :meth:`run` method."""
+        options = super()._default_run_options()
+
+        options.meas_level = MeasLevel.KERNELED
+        options.meas_return = "single"
+
+        return options
+
+    def __init__(
+        self,
+        qubit: int,
+        frequencies: Iterable[float],
+        backend: Optional[Backend] = None,
+        absolute: bool = True,
+        analysis: Optional[BaseAnalysis] = None,
+        **experiment_options,
+    ):
+        """A spectroscopy experiment where the frequency of a pulse is scanned.
+
+        Args:
+            qubit: The qubit on which to run spectroscopy.
+            frequencies: The frequencies to scan in the experiment, in Hz.
+            backend: Optional, the backend to run the experiment on.
+            absolute: Boolean to specify if the frequencies are absolute or relative to the
+                qubit frequency in the backend.
+            analysis: An instance of the analysis class to use.
+            experiment_options: Key word arguments used to set the experiment options.
+
+        Raises:
+            QiskitError: if there are less than three frequency shifts.
+
+        """
+        analysis = analysis or ResonanceAnalysis()
+
+        super().__init__([qubit], analysis=analysis, backend=backend)
+
+        if len(frequencies) < 3:
+            raise QiskitError("Spectroscopy requires at least three frequencies.")
+
+        self._frequencies = frequencies
+        self._absolute = absolute
+
+        self.set_experiment_options(**experiment_options)
+
+    @property
+    @abstractmethod
+    def _backend_center_frequency(self) -> float:
```

Comment:
Can we leverage set backend method for this? i.e. we can save target freq in private `self._center_freq` variable?
https://github.com/Qiskit/qiskit-experiments/blob/a0beb41ec0757429031e0e6e7a7855dcc72877cd/qiskit_experiments/framework/base_experiment.py#L138-L144

Reply:
I think we can leverage `_set_backend` for dt and granularity. Thanks for the suggestions. I did not do this for the center frequency. The reason is that the current implementation points out that the center frequency is something that needs to be taken care of properly. Putting it in `_set_backend` makes this less obvious.